src/address.rs

```diff
@@ -39,6 +43,26 @@ impl Address {
     pub(crate) fn pk_d(&self) -> &DiversifiedTransmissionKey {
         &self.pk_d
     }
+
+    /// Serializes this address to its "raw" encoding as specified in [Zcash Protocol Spec § 5.6.4.2: Orchard Raw Payment Addresses][orchardpaymentaddrencoding]
+    ///
+    /// [orchardpaymentaddrencoding]: https://zips.z.cash/protocol/protocol.pdf#orchardpaymentaddrencoding
+    pub fn to_raw_address_bytes(&self) -> [u8; 43] {
+        let mut result = [0u8; 43];
+        result[..11].copy_from_slice(self.d.as_array());
+        result[11..].copy_from_slice(&self.pk_d.to_bytes());
+        result
+    }
+
+    /// Parse an address from its "raw" encoding as specified in [Zcash Protocol Spec § 5.6.4.2: Orchard Raw Payment Addresses][orchardpaymentaddrencoding]
+    ///
+    /// [orchardpaymentaddrencoding]: https://zips.z.cash/protocol/protocol.pdf#orchardpaymentaddrencoding
+    pub fn from_raw_address_bytes(bytes: &[u8; 43]) -> CtOption<Self> {
+        DiversifiedTransmissionKey::from_bytes(bytes[11..].try_into().unwrap()).map(|pk_d| {
+            let d = Diversifier::from_bytes(bytes[..11].try_into().unwrap());
```

Comment:
All 11-byte strings are valid Orchard diversifiers, but the same is not true for pk<sub>d</sub>, so I don't understand why it's safe to `unwrap` there. Can't this be called with any arbitrary 43-byte string?

Reply:
I think the `unwrap()` here is on the array `try_into()` (and not the `DiversifiedTransmissionKey::from_bytes()`).

Reply:
Yes, you're right.